src/main/java/org/embulk/input/marketo/delegate/ProgramMembersBulkExtractInputPlugin.java

```diff
@@ -0,0 +1,243 @@
+package org.embulk.input.marketo.delegate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.lang3.StringUtils;
+import org.embulk.base.restclient.ServiceResponseMapper;
+import org.embulk.base.restclient.record.RecordImporter;
+import org.embulk.base.restclient.record.ServiceRecord;
+import org.embulk.base.restclient.record.ValueLocator;
+import org.embulk.config.ConfigDiff;
+import org.embulk.config.ConfigException;
+import org.embulk.config.TaskReport;
+import org.embulk.input.marketo.CsvTokenizer;
+import org.embulk.input.marketo.MarketoService;
+import org.embulk.input.marketo.MarketoServiceImpl;
+import org.embulk.input.marketo.MarketoUtils;
+import org.embulk.input.marketo.bulk_extract.AllStringJacksonServiceRecord;
+import org.embulk.input.marketo.bulk_extract.CsvRecordIterator;
+import org.embulk.input.marketo.model.MarketoField;
+import org.embulk.input.marketo.rest.MarketoRestClient;
+import org.embulk.spi.DataException;
+import org.embulk.spi.Exec;
+import org.embulk.spi.PageBuilder;
+import org.embulk.spi.Schema;
+import org.embulk.util.config.Config;
+import org.embulk.util.config.ConfigDefault;
+import org.embulk.util.file.InputStreamFileInput;
+import org.embulk.util.text.LineDecoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.embulk.input.marketo.MarketoInputPlugin.CONFIG_MAPPER_FACTORY;
+
+public class ProgramMembersBulkExtractInputPlugin extends MarketoBaseInputPluginDelegate<ProgramMembersBulkExtractInputPlugin.PluginTask>
+{
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public interface PluginTask extends MarketoBaseInputPluginDelegate.PluginTask, CsvTokenizer.PluginTask
+    {
+        @Config("program_ids")
+        @ConfigDefault("null")
+        Optional<String> getProgramIds();
+
+        @Config("polling_interval_second")
+        @ConfigDefault("60")
+        Integer getPollingIntervalSecond();
+
+        @Config("bulk_job_timeout_second")
+        @ConfigDefault("3600")
+        Integer getBulkJobTimeoutSecond();
+
+        //https://developers.marketo.com/rest-api/bulk-extract/
+        @Max(2)
```

Comment:
Why accept max 2 concurrent batches? I think max 10 can be possible, but should keep default 2

Reply:
because marketo only support 2 export job at same time. Although they support 10 queue jobs (include 2 export job)